### ISSUE TITLE
Exposing embedded search params in ccards

### ISSUE THREAD

**Issue by ibestvina** [xpgo/obsidian-folder-note-plugin#11]
[Embedded search](https://publish.obsidian.md/help/Plugins/Search#Embed+search+results) in Obsidian seems to have a lot of overlap with the live-ccard idea. However, ccard is way better-looking, while embedded search is simply searching through all the notes and listing the results.

Is there a search API available to the plugins? If yes, is it possible to expose embedded search params to ccards? I am asking all this in the context of the following feature request I opened: https://forum.obsidian.md/t/option-to-include-the-first-note-image-in-embeded-query-results/13379

If something like this is possible, I think that enabling it would make the ccard+query a very popular new plugin which enables you to build custom "catalog" notes. It might also simplify the development of this plugin, making new features very simple to add (e.g. add a section of cards which are tagged with the current note-folder name, or are linking to it, etc.)

**Comment by xpgo** [comment]
It seems there are a lot things to do with the ccard. At the begining, I did not expect so many features with ccard. 
I think this is a very good suggestion, however, it needs much time to reconsider the issure and related requirements, or the syntax of ccard will be out of control. 

As you said, the search query is for data, ccard is for UI. There may be an elegent way to integrate them. I will think about it when have time. Thanks.

**Comment by ibestvina** [comment]
I agree with the careful approach, it can easily blow out of control yes. Thanks for quick responses!
